File: twoPointers/inward/pairSum.js
function pair_sum_sorted(nums, target) {
    const answer = [];
    let start = 0;
    let end = nums.length - 1;

    while (start < end) {
        let sum = nums[start] + nums[end];
        if (sum === target) {
            answer.push(start, end);
            break
        }
        else if (sum > target) {
            end = end - 1;
        }
        else {
            start = start + 1;
        }
    }
    return answer;
}


const nums = [1, 2, 3, 4, 5, 6];
const target = 7;
const result = pair_sum_sorted(nums, target);
console.log("Pairs:", result);

File: twoPointers/cpp/inward/pairSum.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> pairSum(vector<int> nums, int target){
    int start = 0;
    int end = nums.size() - 1;
    vector<int> ans;
    
    while(start < end){
        int sum = nums[start] + nums[end];

        if(sum < target){
            start++;
        }
        else if(sum > target){
            end--;
        }
        else{
            ans.push_back(start);
            ans.push_back(end);
            break;
        }
    }

    return ans;
}

int main(){
    vector<int> nums = {-5,-2,3,4,6};
    int target = 7;
    vector<int> ans = pairSum(nums, target);
    for(auto x : ans){
        cout<<x<<" ";
    }
}